### PR TITLE
Fixed #203, doctests never failing

### LINKED ISSUE #203: doc tests are never failing
@avoinea @alecghica 

this is due to https://github.com/eea/eea.facetednavigation/blob/master/eea/facetednavigation/tests/test_doctests.py#L25

Gauthier

### PR DESCRIPTION
**Pull request by gbastien**
Hi @avoinea 

I fixed Py23DocChecker check_output and that broke only one test unicode.txt

Testing in the browser, when value is a unicode, it is set in self.request.form as utf8

Actually, by default query does not receive any kwargs, it could be done while calling @@faceted_query and passing parameters, but in this case, it is received as utf-8 as well...  And I do not see how this could be done, when @@faceted_query is called from browser with parameters, it is passed in request.form and not in kwargs...

Anyway, I think this is correct.

Gauthier

### CODE REVIEW

**Comment by avoinea** [comment]
Interesting `Plone4` fail:
```
File "/plone/instance/src/eea.facetednavigation/eea/facetednavigation/docs/language.txt", line 34, in language.txt
Failed example:
    controllang.available_languages
Expected:
    [u'en']
Got:
    [u'en']
```

And `Python3`:
```
----------------------------------------------------------------------

File "/plone/instance/src/eea.facetednavigation/eea/facetednavigation/docs/unicode.txt", line 66, in unicode.txt
Failed example:
    for brain in brains:
        print(brain.Title)
Expected:
    NON-ASCII Folder
Got nothing
```

**Comment by gbastien** [comment]
@avoinea Py3 tests are failing in the unicode.txt file I adapted...  I will check this to avoid use encode('utf-8')...

**Comment by gbastien** [comment]
@avoinea my change in unicode.txt breaks py3 and fixes py2...

**Comment by gbastien** [comment]
@avoinea ok the only test still broken is language.txt with this weird error...

`Expected:
    [u'en']
Got:
    [u'en']`

Maybe it is related with the manipulation done in check_output?

File: eea/facetednavigation/docs/unicode.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ Add some content
     >>> doc = sandbox.invokeFactory('News Item', 'non-ascii-folder', title='NON-ASCII Folder')
     >>> doc = sandbox._getOb(doc)
     >>> doc.setLanguage('en')
-    >>> doc.setSubject((u'\u015etire',))
+    >>> doc.setSubject(('$tire',))
```
Comment (avoinea) [inline eea/facetednavigation/docs/unicode.txt:40]: :) That should actually be `Ș`. Would be also ok to replace it with any non-ascii french char

Comment (gbastien) [reply eea/facetednavigation/docs/unicode.txt]: Sure, I replaced it with "é", moreover $ is an ascii char so the test was completely missing it's target...

Jenkins should be running again...

**Comment by gbastien** [comment]
Hi @avoinea 

I do not have easy deploy of env with LinguaPlone that I do not use at all...

If you have time, please can you have a look, moreover the error is really not obvious, I would just remove this test as we change the value to en/fr just under but ;-)

**Comment by gbastien** [comment]
@avoinea still failing with the same originality...

**Comment by gbastien** [comment]
Hi @avoinea 

I do not get why https://github.com/eea/eea.facetednavigation/pull/204/commits/19af05426881de8001287ee265c72c12acd0a60a is breaking widget.txt tests...

Anyway, when removing the Py23DocChecker for language.txt, tests pass, that would mean that it is related to string manipulation...

**Comment by gbastien** [comment]
Tests are passing, wouhouh ;-)

**Review by avoinea — APPROVED** [review]
LGTM.

**Pull request merged**